plugins/fabric8-end2end-flow/src/main/resources/end2end/files/RhCheKeycloak.js

```diff
@@ -244,7 +206,7 @@ function initAnalytics(writeKey){
         return new Promise((resolve, reject) => {
             var request = new XMLHttpRequest();
             request.onerror = request.onabort = function(error) {
-                reject(error);
+                reject(this);
```

Comment:
it's unsual to reject this ( usually it's only errors)

Reply:
Well, in the `get` and `post`, when an error occurs (or a non-successful status code), I just return the request itself. This way, in the calling code, I can extract whatever information I need from the request.
Is it a problem ?